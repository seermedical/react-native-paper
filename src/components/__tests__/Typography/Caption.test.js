import * as React from 'react';
import { StyleSheet } from 'react-native';
import renderer from 'react-test-renderer';
import Caption from '../../Typography/Caption.tsx';
import { red500 } from '../../../styles/colors';

const style = StyleSheet.create({
  caption: {
    fontSize: 20,
    color: red500,
  },
});

it('renders caption with children as content', () => {
  const tree = renderer.create(<Caption>Caption content</Caption>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders caption applying style', () => {
  const tree = renderer
    .create(<Caption style={style.caption}>Big and red caption</Caption>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
