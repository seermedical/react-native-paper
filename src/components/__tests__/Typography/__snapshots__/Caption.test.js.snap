// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders caption applying style 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      Object {
        "textAlign": "left",
      },
      Array [
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "color": "#f44336",
            "fontSize": 20,
          },
        ],
      ],
    ]
  }
>
  Big and red caption
</Text>
`;

exports[`renders caption with children as content 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      Object {
        "textAlign": "left",
      },
      Array [
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          undefined,
        ],
      ],
    ]
  }
>
  Caption content
</Text>
`;
